src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs

```diff
@@ -76,8 +76,7 @@ private void OnError(int? contextId, string error)
             int contextId;
             if (!contexts.TryGetValue(projectPath, out contextId))
             {
-                // This should never happen
-                throw new InvalidOperationException();
+                throw new InvalidOperationException($"Failed to identify the context of a project at {projectPath}. It could be caused by multiple global.json introduced different version of sdk for the same project.");
```

Comment:
I am thinking about a better message. This "Failed to identify the context" is quite cryptic.